clients/fides-js/src/lib/consent.tsx

```diff
@@ -0,0 +1,305 @@
+            if (typeof value[1] !== "string") {
+                debugLog(`Invalid banner options: labels.${value[0]} is not a string!`);
+                validLabels = false;
+            }
+        });
+
+        if (!validLabels) {
+            return false;
+        }
+    }
+
+    if (!options.privacyCenterUrl) {
+        debugLog("Invalid banner options: privacyCenterUrl is required!");
+        return false;
+    }
+
+    if (options.privacyCenterUrl) {
+        try {
+            // eslint-disable-next-line no-new
+            new URL(options.privacyCenterUrl);
+        } catch (e) {
+            debugLog("Invalid banner options: privacyCenterUrl is an invalid URL!", options);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+/**
+ * Fetch the user's geolocation from an external API
+ */
+const getLocation = async (): Promise<UserGeolocation> => {
+    debugLog("Running getLocation...");
+    const geolocationApiUrl = "http://dev-cdn-api.ethyca.com/location"; // TODO: replace with prod
+
+    debugLog(`Calling geolocation API: GET ${geolocationApiUrl}...`);
+    const fetchOptions: RequestInit = {
+        mode: "cors"
+    };
+    const response = await fetch(geolocationApiUrl,fetchOptions);
+
+    if (!response.ok) {
+        debugLog("Error getting location from geolocation API, returning {}. Response:", response);
+        return {};
+    }
+
+    try {
+        const body = await response.json();
+        debugLog("Got location response from geolocation API, returning:", body);
+        return body;
+    } catch (e) {
+        debugLog("Error parsing response body from geolocation API, returning {}. Response:", response);
+        return {};
+    }
+}
+
+/**
+ * Determine whether or not the banner should be enabled for the given location
+ */
+const isBannerEnabledForLocation = (location?: UserGeolocation): boolean => {
+    const options = getBannerOptions();
+    if (location === undefined || !location) {
+        debugLog("Location unknown, assume banner must be shown. isBannerEnabledForLocation = true");
+        return true;
+    }
+
+    // Get the user's country
+    if (location.country === undefined || !location.country) {
+        debugLog("Country unknown, assume banner must be shown. isBannerEnabledForLocation = true");
+        return true;
+    }
+
+    if (options.isEnabledCountries) {
+        if (options.isEnabledCountries.includes(location.country)) {
+            debugLog(`Country ${location.country} included in isEnabledCountries, banner must be shown. isBannerEnabledForLocation = true`);
+            return true;
+        }
+        debugLog(`Country ${location.country} not included in isEnabledCountries, banner must be hidden. isBannerEnabledForLocation = false`);
+        return false;
+    }
+
+    debugLog("No location-specific rules matched, assume banner must be shown. isBannerEnabledForLocation = true");
+    return true;
+}
+
+const defaultStyle = `:root {
```

Comment:
Duplicated from banner.css, can this one get removed?

Reply:
Oh right, I still haven't gotten standalone css loading properly using tsup. I'll aim for this today/tomorrow